### PR TITLE
Implement LWG-3683: `operator==` for `polymorphic_allocator` cannot deduce template argument in common cases

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
Also applying this to C++17 mode.

Towards #2872.

### CODE REVIEW

File: stl/inc/xpolymorphic_allocator.h [hunk 1]
```diff
@@ -282,6 +282,18 @@ namespace pmr {
             return _Resource;
         }
 
+        _NODISCARD_FRIEND bool operator==(
```
Comment (strega-nil-ms) [inline stl/inc/xpolymorphic_allocator.h:285]: Given that this is not in the actual C++20 standard, I would prefer to see a reference to `LWG-3683` here.

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]

File: stl/inc/xpolymorphic_allocator.h (hunk 1 above)
Comment (frederick-vs-ja) [reply stl/inc/xpolymorphic_allocator.h:285]: It seems that MSVC STL tends to reference LWG issues in standard headers only if their resolution haven't been accepted. When the issue is officially resolved, STL tends to remove the related comments.

Comment (strega-nil-ms) [reply stl/inc/xpolymorphic_allocator.h:285]: mmh, okay. I don't like this policy but given that it is policy, I'm good with it.

**Review by strega-nil-ms — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for implementing so many LWG issue resolutions and improving conformance! :heart_eyes_cat:

**Pull request merged**